### ISSUE TITLE
PB with library and Raspberry Pi

### ISSUE THREAD

**Issue by BomberX** [aws/aws-iot-device-sdk-python#17]
Hello

Your BasicPubSub.py sample file does not work with Raspberry Pi (latest Raspbian version and components), while it works under Windows 10 / Python 2 & 3.
I got a timeout during the mutual authentication (see the traces from another user in the support thread https://forums.aws.amazon.com/thread.jspa?threadID=236865&tstart=0 ).
Has anyone ever succeeded in running this library from a RPI platform ?

**Comment by chaurah** [comment]
Hi @BomberX,
       We have tested the samples and they seem to work fine on Raspberry Pi. Have you tried running the samples with a higher value for Connect timeout? Another option would be to test the device on a faster network. Please try these and let us know if you are able to get a stable connection. Thank you for using AWS IoT.

Rahul

**Comment by liuszeng** [comment]
Hi @BomberX ,

Thank you very much for using AWS IoT Python SDK!

Can you confirm the following things on your Raspberry Pi?

a. Your endpoint is in format \<RANDOM_STRING\>.iot.\<REGION\>.amazonaws.com. If you are using X.509 certificate based mutual authentication, your port should be 8883.
b. Make sure there is no firewall/DNS issue. To verify that, use the following commands in a terminal:

    sudo apt-get install telnet
    telnet <RANDOM_STRING>.iot.<REGION>.amazonaws.com 8883

And if TCP connection is successfully established, the response should be like below:

    Connected to <SOME ENDPOINT>
    Escape character is '^]'.

c. Your Python OpenSSL version should be 1.0.1+. To check that, use the following commands in a interpreter:

    >>> import ssl
    >>> ssl.OPENSSL_VERSION

d. Your credentials are valid and with proper policy/permissions attached to it. You should be able to use the same certificate to connect to AWS IoT using other MQTT client, e.g., [MQTT.fx](http://mqttfx.jfx4ee.org/).

Also, can you provide any debug information you saw when you had the issue?

Thanks,
Liusu

**Comment by BomberX** [comment]
@chaurah @liuszeng 
Thanks for your quick answers !
I finally made it work. 
The only change I made was to restart my Pi3... I didn't think it was a requirement after having installed the python libs, but obviously, it was... Or may be all the mosquitto components I installed on it at the same time required a restart. By the way, I could verify that the AWS service was ok using the mosquitto_sub client from the pi3 before I restarted. Only your python client sample wasn't...
Anyway, it works great now, thank for the assistance anyway.
